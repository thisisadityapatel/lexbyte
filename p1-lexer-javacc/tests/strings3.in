// This is by far the trickiest test, especially the complicated string on line 9!
"Test that escape characters are properly handled"
"A tab between parentheses: (\t) "
"A return between parentheses: (\n) "
"A backslash between parentheses: (\\) "
"A string without tabs or returns between parentheses: (\\t\\n) "
"A quote between parentheses: (\") "
"3 quotes between parentheses: (\"\"\") "
"Complicated string: \n\t\"\" \\\n\\t\\\"\\\\"
"This string should print normally."
