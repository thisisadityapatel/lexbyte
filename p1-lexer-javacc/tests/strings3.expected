String scanned: ["Test that escape characters are properly handled"]
	->Token kind: <STRING>
	->Token value: "Test that escape characters are properly handled"
String scanned: ["A tab between parentheses: (\t) "]
	->Token kind: <STRING>
	->Token value: "A tab between parentheses: (	) "
String scanned: ["A return between parentheses: (\n) "]
	->Token kind: <STRING>
	->Token value: "A return between parentheses: (
) "
String scanned: ["A backslash between parentheses: (\\) "]
	->Token kind: <STRING>
	->Token value: "A backslash between parentheses: (\) "
String scanned: ["A string without tabs or returns between parentheses: (\\t\\n) "]
	->Token kind: <STRING>
	->Token value: "A string without tabs or returns between parentheses: (\t\n) "
String scanned: ["A quote between parentheses: (\") "]
	->Token kind: <STRING>
	->Token value: "A quote between parentheses: (") "
String scanned: ["3 quotes between parentheses: (\"\"\") "]
	->Token kind: <STRING>
	->Token value: "3 quotes between parentheses: (""") "
String scanned: ["Complicated string: \n\t\"\" \\\n\\t\\\"\\\\"]
	->Token kind: <STRING>
	->Token value: "Complicated string: 
	"" \
\t\"\\"
String scanned: ["This string should print normally."]
	->Token kind: <STRING>
	->Token value: "This string should print normally."
